transmission_interface/include/transmission_interface/transmission_info.hpp

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TRANSMISSION_INTERFACE__TRANSMISSION_INFO_HPP_
+#define TRANSMISSION_INTERFACE__TRANSMISSION_INFO_HPP_
+
+#include <string>
+
+#include "transmission_interface/visibility_control.h"
+
+namespace transmission_interface
+{
+
+enum class TRANSMISSION_INTERFACE_PUBLIC_TYPE JointControlType: std::uint8_t
```

Comment:
I would like to avoid this enum to ever exist with `ros2_control`. It cements the entire framework into the mindset of only these three interfaces being available and leaves no room for client-side extendability. That being said, I don't yet have a proper solution for this but I'm working on one, this is the reason why #32 is assigned to me